src/views/PasswordReset/index.js

```diff
@@ -0,0 +1,198 @@
+import VisibilityOff from '@material-ui/icons/VisibilityOff';
+import { createMuiTheme } from '@material-ui/core';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
+import Pattern from '../../shared/styles/Pattern';
+import {useStyles} from '../../shared/styles/SignupStyles';
+import Footer from '../../components/Footer/index';
+import Header from '../../components/Header/index';
+import { SignupContext } from '../../contexts/SignupContext';
+import { SET_SIGNUP_LOADING} from '../../actions/types';
+import toaster from '../../helpers/toasts';
+import { toast, ToastContainer, Zoom } from 'react-toastify';
+import { useHistory } from 'react-router-dom';
+import { useTranslation } from "react-i18next";
+
+
+
+const { REACT_APP_BACKEND_URL } = process.env;
+
+const PasswordReset = () => {
+
+  const classes = useStyles();
+  const theme = createMuiTheme();
+  const history = useHistory()
+  const { dispatch } = useContext(SignupContext);
+  const { t } = useTranslation();
+
+
+  const matches = useMediaQuery(theme.breakpoints.up('md'));
+  
+  const [values, setValues] = useState({
+    password: '',
+    newpassword:'',
+    showPassword: false,
+    shownewPassword: false,
+    passwordStrengthVisibility:{display:'none'}
+  });
+  
+  const handleChange = (prop) => (event) => {
+    setValues({ ...values, [prop]: event.target.value });
+  };
+    
+  const handleClickShowPassword = () => {
+    setValues({ ...values, showPassword: !values.showPassword });
+  };
+  const handleClickShownewPassword = () => {
+    setValues({ ...values, shownewPassword: !values.shownewPassword });
+  };
+
+  const handleMouseDownPassword = (event) => {
+    event.preventDefault();
+  };
+  const handleMouseDownnewPassword = (event) => {
+    event.preventDefault();
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+
+    const incomingToken = urlParams.get('token');
+
+    console.log(incomingToken);
+
+    const { password, newpassword } = values;
+
+    dispatch({type: SET_SIGNUP_LOADING});
+    if (password === '') {
+      toaster(t('Password is required'), 'warn')
+      return false;
+    } else if (newpassword === '') {
+      toaster(t('Please confirm the new password'), 'warn')
+      return false;
+    }else if (newpassword.length < 8) {
+      toaster(t('Invalid Password'), 'warn')
+      setValues({ ...values, passwordStrengthVisibility: {display:'block'} });
+      return false;
+    }else if (newpassword !== password) {
+      toaster(t('Password do not match'), 'warn')
+      return false;
+    }
+    else {
+      const result = await fetch(`${REACT_APP_BACKEND_URL}/user/resetnewpassword`, 
+      {
+        mode:'no-cors',
+        method:'POST',
+        headers:{"Content-Type":"Application/json"},
+        body:JSON.stringify({password, token:incomingToken})
```

Comment:
The update endpoint is not post request rather is `Patch` and also consider sending `Accept-Language` in headers just to support multiple language

Reply:
Thank you I need to work on it .

Reply:
Now I have responded to those requested changes.